hasher-matcher-actioner/hmalib/common/actioner_models.py

```diff
@@ -29,39 +29,35 @@ def __eq__(self, another_label: object) -> bool:
             return NotImplemented
         return self.key == another_label.key and self.value == another_label.value
 
-    def __hash__(self) -> int:
-        return self.value.__hash__()
-
-
-class LabelWithConstraints(Label):
-    _KEY_CONSTRAINT = "KeyConstraint"
-
-    def __init__(self, value: str):
-        super(LabelWithConstraints, self).__init__(self._KEY_CONSTRAINT, value)
 
+@dataclass(unsafe_hash=True)
```

Comment:
```suggestion
@dataclass
```

Ignorable/YMMV: You can safely drop the hash generation here, because the number of fields hasn't changed, and so the same hash method will be generated

Reply:
tried this and it didn't work. Discussed further with David and leaving as is.

Reply:
It turns out this advice was wrong! python hash() is smart enough to notice you extended the class without redefining hash, and so hash might not be safe any more.